channels/rdpdr/client/rdpdr_capabilities.c

```diff
@@ -60,12 +61,21 @@ static void rdpdr_write_general_capset(rdpdrPlugin* rdpdr, wStream* s)
 }
 
 /* Process device direction general capability set */
-static void rdpdr_process_general_capset(rdpdrPlugin* rdpdr, wStream* s)
+static UINT rdpdr_process_general_capset(rdpdrPlugin* rdpdr, wStream* s)
 {
 	UINT16 capabilityLength;
 
+	if (!Stream_EnsureRemainingCapacity(s, 2))
```

Comment:
It's Stream_GetRemainingLength here and in line below, and then you should return something to say it's bad data (can't find back the error code)